plugin/kubernetes/node.go

```diff
@@ -0,0 +1,47 @@
+package kubernetes
+
+import (
+	"github.com/coredns/coredns/plugin/etcd/msg"
+	"github.com/coredns/coredns/request"
+	"github.com/miekg/dns"
+	"strings"
+)
+
+func (k *Kubernetes) Node(state request.Request) ([]msg.Service, int) {
+	node := state.Name()[:len(state.Name())-1]
+	nodeList := k.APIConn.NodeIndex(node)
+	services := []msg.Service{}
+	zonePath := msg.Path(state.Zone, coredns)
+	rcode := dns.RcodeNameError
+
+	for _, n := range nodeList {
+		if node != n.Name {
+			continue
+		}
+
```

Comment:
Returning all internal and external IPs doesn't feel like the right thing to do here.  Seems like it should return internal or external, but not both.

Reply:
Agree. It could return externalIP if it exists, and if not the internal IP

Reply:
Hmm. Everything else returned by cluster DNS is internal. If the externalIP is firewalled (highly likely), this could cause issues. Refer to https://github.com/kubernetes/kubernetes/issues/90764

If you want to serve both you could have a separate external zone, perhaps?